File: persistence/src/main/java/org/devgateway/toolkit/persistence/dto/GTPLocation.java
package org.devgateway.toolkit.persistence.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import org.devgateway.toolkit.persistence.dao.location.Department;

/**
 * @author Nadejda Mandrescu
 */
@JsonInclude(JsonInclude.Include.ALWAYS)
public class GTPLocation {

    private final Long id;
    private final String name;

    public GTPLocation(Department department) {
        this.id = department.getId();
        this.name = department.getName();
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: ui20/src/modules/entities/bulletins/GTPLocation.js
export default class GTPLocation {
  id: number
  name: String

  constructor({id, name} = {}) {
    this.id = id
    this.name = name
  }

  static localeCompare(l1: GTPLocation, l2: GTPLocation) {
    if (l1.id === null) return -1
    if (l2.id === null) return 1
    return l1.name.localeCompare(l2.name)
  }
}
